python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,72 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args
+    ) -> "Dataset[Any]":
+        """Apply a function to each group of records.
+
+        The user given function should take as argument a batch, which contains
+        exactly the records for a group, and return a block of the same batch type.
+        The results of different groups will be collected to form the result dataset.
+        """
+
+        # Globally sort records by key.
+        # Note that sort() will ensure that records of the same key partitioned
+        # into the same block.
```

Comment:
I think `sort() will ensure that records of the same key partitioned into the same block.` is not part of the sort public api but rather implementation detail and the assumption may break in the future without notice. 

Can we at least add some sort tests to check this assumption so that if sort behavior changes in the future, we are able to catch it. Also adding some comment to `sort_impl` saying that map_groups relies on the implementation to partition the same key to the same block?

Reply:
Agreed, we also chatted about this in the meeting on the `.map_groups()` API.

Reply:
I definitely agree it's not great to relying on something that's not contracted by sort() API.
Your suggestion sounds good: I added comments to sort_impl and unit tests for sort().
PTAL, thanks.

Reply:
> Lint error

Fixed. Also ran scripts/format.sh.